src/AbstractActionHandler.test.ts

```diff
@@ -236,7 +236,120 @@ describe('Action Handler', () => {
     expect(runEffectAfterUpgrade).toHaveBeenCalledTimes(1)
   })
 
-  it('continues ifSetup is true', async () => {
+  it('rolls back', async () => {
+    const blockMeta1 = {
```

Comment:
Something for future consideration is if there's a fixtures library/tool we want to use to replace this kind of setup. (Or just doing the simple `import ./testhelpers.ts` kind of thing)

Reply:
I considered writing a function for this, but I do need control over all the inputs, so I don't know if it would actually make it less verbose. I could make defaults, but I'm hesitant to go down that road since there's already more surface areas for bugs in these tests than I'm comfortable with, and I don't want to contribute to the problem.